activerecord/lib/active_record/query_logs.rb

```diff
@@ -99,10 +99,12 @@ def update_context(**options)
       # execution of the provided block. Resets provided values to nil after
       # the block is executed.
       def set_context(**options)
+        keys = options.keys
+        previous_context = keys.zip(context.values_at(*keys)).to_h
```

Comment:
Can we call slice on the context and pass in the options keys instead?

Reply:
That was my first thought, but `slice` won't include missing keys.